core/src/main/scala/kafka/server/MetadataSupport.scala

```diff
@@ -56,11 +59,24 @@ sealed trait MetadataSupport {
    */
   def ensureConsistentWith(config: KafkaConfig): Unit
 
-  def maybeForward(request: RequestChannel.Request,
-                   handler: RequestChannel.Request => Unit,
-                   responseCallback: Option[AbstractResponse] => Unit): Unit
+  /**
+   * Get the dynamic configuration associated with a given resource.
+   */
+  def getResourceConfig(resource: ConfigResource): Properties
```

Comment:
The implementations for this method seems to be the same as `ConfigResource::config` can we figure out a way to rely on that implementation? Or remove the code duplication?

Reply:
Yes, this is a duplicate of the code in `ConfigRepository`. I'll just use that.